.github/workflows/performance-regression.yml

```diff
@@ -1,152 +1,140 @@
           do
-            java -jar $jar_file generate -S 50000 --input-ion-schema $schema_dir/${test}.isl testData/${test}.10n
+            java -jar $jar_file generate -S ${{env.data_size}} --input-ion-schema $schema_dir/${test}.isl testData/${test}.10n
           done
-
       - name: Upload test Ion Data to artifacts
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
-          name: test Ion Data
+          name: ${{env.test_data_id}}
           path: testData
 
-      # Generates performance results for the current commit
-      - name: Checkout the current commit of the Ion Python
-        uses: actions/checkout@v3
+  prepopulate-pip-cache:
+    # Since all the "Check" jobs can run in parallel, caching _could_ have basically no effect. In order to speed things
+    # up, this step can run in parallel with the "Generate Data" job, pre-caching all the dependencies.
+    name: Setup PIP Cache
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
-          ref: ${{ github.event.pull_request.head.sha }}
-          path: ion-python-current
-
-      - name: Create a virtual environment and setup the package
-        working-directory: ./ion-python-current
-        run: |
-          git submodule init && git submodule update && python3 -m venv ./venv && . venv/bin/activate
+          python-version: '3.10'
+          cache: 'pip'
+          cache-dependency-path: '**/requirements.txt'
+      - run: |
           pip install -r requirements.txt
-          pip install -e .
-
-      - name: Running performance benchmark
-        working-directory: ./ion-python-current
-        env:
-          warmups: 100
-          iterations: 1000
-          cli_script: amazon/ionbenchmark/ion_benchmark_cli.py
-          runner_dir: /Users/runner/work/ion-python/ion-python
-        run: |
-          . venv/bin/activate
-          for testset in nestedStruct nestedList sexp realWorldDataSchema01 realWorldDataSchema02 realWorldDataSchema03
-          do
-            python $cli_script read -w $warmups --iterations $iterations -o $runner_dir/benchmarkResults/${testset}_read/new --io-type buffer --format ion_text --format ion_binary --api load_dump $runner_dir/testData/${testset}.10n
+          # TODO: See if there's a way to cache the ion-c build output if it hasn't changed
 
-            python $cli_script write -w $warmups --iterations $iterations -o $runner_dir/benchmarkResults/${testset}_write/new --io-type buffer --format ion_text --format ion_binary --api load_dump $runner_dir/testData/${testset}.10n
-          done
+  detect-regression:
+    name: Check
+    runs-on: ubuntu-latest
+    needs: [generate-test-data, prepopulate-pip-cache]
+    strategy:
+      matrix:
+        python-version: ['3.9', '3.11', 'pypy-3.8', 'pypy-3.10']
```

Comment:
I'm thinking we should just test against the most recent supported version of both CPython and PyPy.

Reply:
I created #286 for this discussion so that it doesn't have to block this PR.